src/application/Application.mjs

```diff
@@ -103,11 +103,22 @@ export default class Application extends Component {
     }
 
     __updateFocus() {
+        const prevFocusedComponent = this._focusPath ? this._focusPath[this._focusPath.length - 1] : undefined;
         const notOverridden = this.__updateFocusRec();
 
         if (!Application.booting && notOverridden) {
             this.updateFocusSettings();
         }
+
+        if (this.__options.debug) {
+            const newFocusedComponent = this._focusPath ? this._focusPath[this._focusPath.length - 1] : undefined;
+            prevFocusedComponent && prevFocusedComponent.updateDebugFocusState && prevFocusedComponent.updateDebugFocusState();
+            newFocusedComponent && newFocusedComponent.updateDebugFocusState && newFocusedComponent.updateDebugFocusState();
+
+            if (prevFocusedComponent !== newFocusedComponent) {
+                console.log('FOCUS ' + (newFocusedComponent && newFocusedComponent.getLocationString()));
```

Comment:
Moving this here respects the original intent better; it will now state the change in focus on _initial_ focus in the app (previously would only do it for subsequent changes).